File: MVC_DiceGame/MVC_DiceGame_java/view/SweConsole.java
package view;

public class SweConsole implements IView {
  private int getInputChar() {
    try {
      int c = System.in.read();
      while (c == '\r' || c =='\n') {
        c = System.in.read();
      }
      return c;
    } catch (java.io.IOException e) {
      System.out.println("" + e);
      return 0;
    }
  }
  
  public boolean wantsToPlay() {
    return getInputChar() != 'a';
  }
  
  public void DisplayResult(boolean a_win, int a_dice1Val, int a_dice2Val) {
    if (a_win) {
      System.out.println("Vann: " + a_dice1Val + " " + a_dice2Val);
    } else {
      System.out.println("Förlorade:" +a_dice1Val + " " + a_dice2Val);
    }
  }
  
  public void DisplayWelcomeMessage() {
    System.out.println("Välkommen till tärningsspelet");
  }
  
  public void DisplayInstructions() {
    System.out.println("Truck en tangent för att spela eller 'a' för att avsluta");
  }
}

File: MVC_DiceGame/MVC_DiceGame_Swift_3.0.1/MVC_DiceGame_Swift_3.0.1/view/SweConsole.swift
//
//  SweConsole.swift
//  MVC_DiceGame_Swift_3.0.1
//
//  Created by Sarpreet Singh on 2016-11-13.
//  Copyright © 2016 Sarpreet Singh. All rights reserved.
//

import Foundation

public class SweConsole : IView {
    
    func getInputChar() -> String {
        var input : String = readLine()!;
        while input == "\n" || input == "\r" || input.characters.count > 1{
            input = readLine()!;
        }
        return input;
    }
    
    func wantsToPlay() -> Bool {
        return getInputChar() != "a";
    }
    
    func displayResult(a_win : Bool, a_dice1Val : Int, a_dice2Val : Int) {
        if a_win {
            print("Vann: " + String (a_dice1Val) + "  " + String (a_dice2Val));
        } else {
            print("Förlorade: " + String (a_dice1Val) + "  " + String (a_dice2Val));
        }
    }
    
    func displayWelcomeMessage() {
        print("Välkommen till tärningsspelet");
    }
    
    func displayInstructions() {
        print("Truck en tangent för att spela eller 'a' för att avsluta");
    }
}
